Fixed part of #8472: Migrated create-new-story to angular component

ShivamJhaa opened the pull request:
## Overview
<!--
READ ME FIRST:
Please answer *both* questions below and check off every point from the Essential Checklist!
If there is no corresponding issue number, fill in N/A where it says [fill_in_number_here] below in 1.
-->

1. This PR fixes or fixes part of #8472.
2. This PR does the following: Migrated create-new-story to angular component

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## Proof that changes are correct

https://user-images.githubusercontent.com/86483059/145049594-6ef42765-c7b5-47ed-a844-e850fedbcd16.mp4





<!--
Add videos/screenshots of the user-facing interface in various display sizes (mainly phone, tablet, and desktop display size) to demonstrate that the changes made in this PR work correctly.
If the changes in your PRs are autogenerated via a script and you cannot provide proof for the changes then please leave a comment "No proof of changes needed because {{Reason}}".
-->

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change).
- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.
- Oppiabot can assign anyone for review/help if you leave a commen



t like the following: "{{Question/comment}} @{{reviewer_username}} PTAL"
- Some of the e2e tests are flaky, and can fail for reasons unrelated to your PR. We are working on fixing this, but in the meantime, if you need to restart the tests, please check the ["If your build fails" wiki page](https://github.com/oppia/oppia/wiki/If-your-build-fails).

ShivamJhaa committed:
Checks

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into develop

ShivamJhaa committed:
update

ShivamJhaa committed:
Fix part of issue #8472: Migrated create-new-story to angular component

ShivamJhaa committed:
updated spec file

ShivamJhaa committed:
Updated spec files

ShivamJhaa committed:
Updated spec files

ShivamJhaa committed:
Updated spec files

ShivamJhaa committed:
Updated spec files

ShivamJhaa committed:
Updated spec files

ShivamJhaa committed:
updated spec files

ShivamJhaa committed:
updated spec files

ShivamJhaa committed:
updated spec files

ShivamJhaa committed:
updated spec files

ShivamJhaa committed:
updated spec files

ShivamJhaa commented:
@EricZLou @kevintab95 @DubeySandeep  PTAL

ShivamJhaa committed:
updated html file

DubeySandeep commented:
@ShivamJhaa Frontend tests are failing on this PR, PTAL!

EricZLou commented on core/templates/components/entity-creation-services/story-creation.service.ts:
```diff
@@ -39,12 +40,8 @@ angular.module('oppia').factory('StoryCreationService', [
-            '/pages/topic-editor-page/modal-templates/' +
-            'create-new-story-modal.template.html'),
-          backdrop: 'static',
-          controller: 'CreateNewStoryModalController'
+        return NgbModal.open(CreateNewStoryModalComponent, {
+          backdrop: true
         }).result.then(function(newlyCreatedStory) {
```
I'm not sure about this, but is `backdrop: 'static'` the same as `backdrop: true`?

ShivamJhaa replied:
No, It was written by mistake.
Updated now

ShivamJhaa committed:
Updated

ShivamJhaa committed:
Updated

ShivamJhaa committed:
Updated

ShivamJhaa committed:
updated

Radesh-kumar commented on core/templates/components/entity-creation-services/story-creation.service.spec.ts:
```diff
@@ -68,62 +75,69 @@ describe('Story Creation Service', () => {
-    $scope.$apply();
-
-    $httpBackend.flush();
+    tick();
```
I think we need `$scope.$apply();` after this.
(Since the service is still in angJS)

ShivamJhaa replied:
It's still not working

Radesh-kumar replied:
<pre>  it('should post story data to server and change window location' +
    ' on success', fakeAsync(() => {
    spyOn(ngbModal, 'open').and.returnValue(
      {
        result: Promise.resolve({
          isValid: () => true,
          title: 'Title',
          description: 'Description',
          urlFragment: 'url'
        })
      } as NgbModalRef
    );

    $httpBackend.expect('POST','/topic_editor_story_handler/id')
      .respond(200, {storyId: 'id'});

    expect(mockWindow.location).toBe('');

    StoryCreationService.createNewCanonicalStory();

    $scope.$apply()
    tick();
    $httpBackend.flush();

    expect(mockWindow.location).toBe('/story_editor/id');
  }));</pre>

ShivamJhaa replied:
Done.

ShivamJhaa committed:
Updated

ShivamJhaa committed:
Updated

ShivamJhaa committed:
Updated

ShivamJhaa committed:
Updated

ShivamJhaa committed:
Fixed frontend test cases

ShivamJhaa committed:
Fixed frontend test cases

EricZLou commented on core/templates/pages/topic-editor-page/modal-templates/create-new-story-modal.component.html:
```diff
@@ -22,55 +26,64 @@ <h3>
-                                [allowed-bg-colors]="allowedBgColors"
-                                [thumbnail-bg-color]="story.getThumbnailBgColor()"
-                                preview-description-bg-color="#2F6687"
-                                (image-save)="updateView()">
```
You need to keep this `updateView` function -- that is probably why the thumbnail is not updating.

aks681 replied:
+1. The video in the description is not the expected result.

EricZLou commented on core/templates/pages/topic-editor-page/modal-templates/create-new-story-modal.controller.ts:
```diff
@@ -1,82 +0,0 @@
-        $scope.story.urlFragment, function() {
-          $scope.storyUrlFragmentExists = (
-            StoryEditorStateService.getStoryWithUrlFragmentExists());
-          $rootScope.$applyAsync();
```
You're missing this `$rootScope.$applyAsync();` in the migrated code above.

EricZLou commented on core/templates/pages/topic-editor-page/modal-templates/create-new-story-modal.component.ts:
```diff
@@ -0,0 +1,86 @@
+      this.newlyCreatedStory.urlFragment, () => {
+        this.storyUrlFragmentExists = (
+          this.storyEditorStateService.getStoryWithUrlFragmentExists());
+      });
```
You need to add a line here to get the URL fragment changes to show up -- see my other comment on `$scope.$apply()`!

EricZLou reviewed (changes requested):
See comments below for responses to your Google chats!

kevintab95 commented:
Could someone from the migration team please take a first pass on this PR if possible? /cc @ashutoshc8101 @srijanreddy98

ashutoshc8101 commented on core/templates/pages/topic-editor-page/editor-tab/topic-editor-tab.directive.spec.ts:
```diff
@@ -20,14 +20,23 @@
 import { EventEmitter } from '@angular/core';
 import { ShortSkillSummary } from 'domain/skill/short-skill-summary.model';
 import { Subtopic } from 'domain/topic/subtopic.model';
+require('services/ngb-modal.service.ts');
```
Are changes to this file required?

ShivamJhaa replied:
Frontend tests are failing without this.

ashutoshc8101 commented on core/templates/pages/topic-editor-page/modal-templates/create-new-story-modal.component.html:
```diff
@@ -1,19 +1,23 @@
 <div class="modal-header create-new-story-header">
   <h3>
-    New Story
+    New story
```
Any reason for this change?

ShivamJhaa replied:
Updated

ashutoshc8101 commented on core/templates/pages/topic-editor-page/modal-templates/create-new-story-modal.component.html:
```diff
@@ -1,19 +1,23 @@
+  <i class="fa fa-times modal-close-button" (click)="cancel()"></i>
 </div>
-<form ng-submit="confirm(story)">
+<form (submit)="save()">
```
As you are inheriting from `ConfirmOrCancelModal`, you should use and override confirm function instead.

ShivamJhaa replied:
Done

ashutoshc8101 commented on core/templates/pages/topic-editor-page/modal-templates/create-new-story-modal.component.html:
```diff
@@ -1,19 +1,23 @@
         <span class="oppia-input-box-subtitle">
           <em>
-            Story title should be at most <[MAX_CHARS_IN_STORY_TITLE]> characters.
+           Story title should be at most {{MAX_CHARS_IN_STORY_TITLE}} characters.
```
Please add space after/before interpolation symbols (`{{` and `}}`)

ShivamJhaa replied:
Done

ashutoshc8101 commented on core/templates/pages/topic-editor-page/modal-templates/create-new-story-modal.component.spec.ts:
```diff
@@ -0,0 +1,131 @@
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { FormsModule } from '@angular/forms';
+import { TopicEditorStateService } from '../services/topic-editor-state.service';
+import constants from 'assets/constants';
```
Please try to use `AppConstants` instead of `constants`.

ShivamJhaa replied:
Done

ashutoshc8101 commented on core/templates/pages/topic-editor-page/modal-templates/create-new-story-modal.component.ts:
```diff
@@ -0,0 +1,86 @@
+  selector: 'oppia-create-new-story-modal',
+  templateUrl: './create-new-story-modal.component.html'
+})
+
```
Can you remove this empty line?

ShivamJhaa replied:
Done

ashutoshc8101 commented on core/templates/pages/topic-editor-page/modal-templates/create-new-story-modal.component.ts:
```diff
@@ -0,0 +1,86 @@
+import { TopicEditorStateService } from '../services/topic-editor-state.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+import { ConfirmOrCancelModal } from 'components/common-layout-directives/common-elements/confirm-or-cancel-modal.component';
+import constants from 'assets/constants';
```
Use `AppConstants` instead.

ShivamJhaa replied:
Done

ashutoshc8101 reviewed (changes requested):
@ShivamJhaa PTAL

ShivamJhaa committed:
Updated as per comments

ShivamJhaa committed:
Updated as per comments

ShivamJhaa committed:
Updated as per comments

ShivamJhaa commented:
@ashutoshc8101 @DubeySandeep  PTAL

ShivamJhaa committed:
Lint checks

ashutoshc8101 commented on core/templates/pages/topic-editor-page/modal-templates/create-new-story-modal.component.ts:
```diff
@@ -0,0 +1,77 @@
+  templateUrl: './create-new-story-modal.component.html'
+})
+export class CreateNewStoryModalComponent extends ConfirmOrCancelModal {
+  allowedBgColors: object = AppConstants.ALLOWED_THUMBNAIL_BG_COLORS.story;
```
Type `object` is too general to use here. You can remove the explicit type from here, the appropriate type will be auto inferred by typescript.

ShivamJhaa replied:
Done

ashutoshc8101 reviewed (changes requested):
Just one nit. Thanks!

srijanreddy98 committed:
Fixes part of #9749: Migrate few instances of angular-html-bind (#14263)

* Revert "Revert "Fix part of #9749: Create a component that shows rte-output-display and use it instead of angular-html-bind. (#13229)" (#13361)"

This reverts commit 792de4596996527a696d0041e0eff2b8384eda30.

* Fix issues

* Fix spelling mistakes in comment

* Merge with upstream/develop

* Address review comments:
  1. Move oppia-rte-output.spec from the list of ignored files for innerHTML check in eslintrc and move to inline disables.
  2. Change the CODEOWNER listing from oppia-rte-parser.service.ts to oppia-rte-parser.service*.ts to include the spec file.
  3. Add refernce to the note at the top of file for all value.replace ops.
  4. Miscellaneous changes.

srijanreddy98 committed:
Fixes part of #9749: Migrate few instances of angular-html-bind (#14263)

* Revert "Revert "Fix part of #9749: Create a component that shows rte-output-display and use it instead of angular-html-bind. (#13229)" (#13361)"

This reverts commit 792de4596996527a696d0041e0eff2b8384eda30.

* Fix issues

* Fix spelling mistakes in comment

* Merge with upstream/develop

* Address review comments:
  1. Move oppia-rte-output.spec from the list of ignored files for innerHTML check in eslintrc and move to inline disables.
  2. Change the CODEOWNER listing from oppia-rte-parser.service.ts to oppia-rte-parser.service*.ts to include the spec file.
  3. Add refernce to the note at the top of file for all value.replace ops.
  4. Miscellaneous changes.

srijanreddy98 committed:
Fixes part of #9749: Migrate few instances of angular-html-bind (#14263)

* Revert "Revert "Fix part of #9749: Create a component that shows rte-output-display and use it instead of angular-html-bind. (#13229)" (#13361)"

This reverts commit 792de4596996527a696d0041e0eff2b8384eda30.

* Fix issues

* Fix spelling mistakes in comment

* Merge with upstream/develop

* Address review comments:
  1. Move oppia-rte-output.spec from the list of ignored files for innerHTML check in eslintrc and move to inline disables.
  2. Change the CODEOWNER listing from oppia-rte-parser.service.ts to oppia-rte-parser.service*.ts to include the spec file.
  3. Add refernce to the note at the top of file for all value.replace ops.
  4. Miscellaneous changes.

Co-authored-by: Vasa Srijan Reddy <redvasa@amazon.com>

ShivamJhaa committed:
Migrated story creation service

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into migration-story-creation

seanlip committed:
Merge branch 'develop' of https://github.com/oppia/oppia into develop

seanlip committed:
Merge branch 'develop' of https://github.com/oppia/oppia into develop

ShivamJhaa committed:
Fixed lint checks

ShivamJhaa committed:
Fixed lint checks

seanlip committed:
Fixes part of #9749: Migrate few instances of angular-html-bind (#14263)

* Revert "Revert "Fix part of #9749: Create a component that shows rte-output-display and use it instead of angular-html-bind. (#13229)" (#13361)"

This reverts commit 792de45.

* Fix issues

* Fix spelling mistakes in comment

* Merge with upstream/develop

* Address review comments:
  1. Move oppia-rte-output.spec from the list of ignored files for innerHTML check in eslintrc and move to inline disables.
  2. Change the CODEOWNER listing from oppia-rte-parser.service.ts to oppia-rte-parser.service*.ts to include the spec file.
  3. Add refernce to the note at the top of file for all value.replace ops.
  4. Miscellaneous changes.

seanlip committed:
Merge branch 'develop' of https://github.com/oppia/oppia into develop

seanlip committed:
Fixes part of #9749: Migrate few instances of angular-html-bind (#14263)
* Revert "Revert "Fix part of #9749: Create a component that shows rte-output-display and use it instead of angular-html-bind. (#13229)" (#13361)"

This reverts commit 792de45.

* Fix issues

* Fix spelling mistakes in comment

* Merge with upstream/develop

* Address review comments:
  1. Move oppia-rte-output.spec from the list of ignored files for innerHTML check in eslintrc and move to inline disables.
  2. Change the CODEOWNER listing from oppia-rte-parser.service.ts to oppia-rte-parser.service*.ts to include the spec file.
  3. Add refernce to the note at the top of file for all value.replace ops.
  4. Miscellaneous changes.

seanlip committed:
Merge branch 'develop' of https://github.com/oppia/oppia into develop

ShivamJhaa committed:
Fixed lint checks

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into migration-story-creation

ShivamJhaa committed:
Merge branch 'develop' of https://github.com/oppia/oppia into migration-story-creation

ShivamJhaa committed:
Removed unwanted change

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into migration-story-creation

srijanreddy98 committed:
Merge branch 'develop' into migration-story-creation

ShivamJhaa committed:
Fixed front-end test

ShivamJhaa committed:
Fixed lint errors

ShivamJhaa committed:
Merge branch 'migration-story-creation' of https://github.com/ShivamJhaa/oppia into migration-story-creation

ShivamJhaa commented:
@DubeySandeep @EricZLou @ashutoshc8101  PTAL

ashutoshc8101 commented on core/templates/components/entity-creation-services/story-creation-backend-api.service.spec.ts:
```diff
@@ -0,0 +1,140 @@
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+
+class MockUrlInterpolationService {
+  interpolateUrl() {
```
Please add return type string.

ShivamJhaa replied:
Done

ashutoshc8101 commented on core/templates/components/entity-creation-services/story-creation-backend-api.service.spec.ts:
```diff
@@ -0,0 +1,140 @@
+      ]
+    });
+
+    csrfService = TestBed.get(CsrfTokenService);
```
`TestBed.get` is deprecated. Try to prefer `TestBed.inject` instead here.

ShivamJhaa replied:
Done

ashutoshc8101 commented on core/templates/components/entity-creation-services/story-creation-backend-api.service.spec.ts:
```diff
@@ -0,0 +1,140 @@
+      imageBlob: imageBlob
+    }];
+
+    // This throws "Argument of type '() -> Promise<unknown>'
+    // is not assignable to parameter of type 'PromiseLike<string>'.".
+    // We need to suppress this error because we need to mock the
+    // `getTokenAsync` function for testing purposes.
+    // @ts-expect-error
+    spyOn(csrfService, 'getTokenAsync').and.returnValue(async() => {
+      return new Promise((resolve) => {
+        resolve('sample-csrf-token');
+      });
+    });
```
This section is not required.

ShivamJhaa replied:
Removed

ashutoshc8101 commented on core/templates/components/entity-creation-services/story-creation-backend-api.service.ts:
```diff
@@ -0,0 +1,93 @@
+// limitations under the License.
+
+/**
+ * @fileoverview Service to notify about creation of topic and obtain
```
Please update the fileoverview.

ShivamJhaa replied:
Done

ashutoshc8101 commented on core/templates/components/entity-creation-services/story-creation-backend-api.service.ts:
```diff
@@ -0,0 +1,93 @@
+  STORY_CREATOR_URL_TEMPLATE: string = '/topic_editor_story_handler/<topic_id>';
+
+  _createStory(
+      successCallback: (value: StoryCreationResponse) => void,
+      errorCallback: (reason: string) => void,
```
Usually `successCallback` and `errorCallback` are placed at the end of parameter list.

ShivamJhaa replied:
Done

ashutoshc8101 commented on core/templates/services/angular-services.index.ts:
```diff
@@ -386,6 +386,9 @@ import { I18nService } from 'i18n/i18n.service';
 import { HistoryTabBackendApiService } from 'pages/exploration-editor-page/services/history-tab-backend-api.service';
+import { StoryCreationService } from 'components/entity-creation-services/story-creation.service';
+import { StoryCreationBackendApiService } from 'components/entity-creation-services/story-creation-backend-api.service';
+
```
Please remove the unnecessary line added.

ShivamJhaa replied:
Done

ashutoshc8101 reviewed (changes requested):
@ShivamJhaa PTAL

ShivamJhaa committed:
Resolved comments

ShivamJhaa committed:
Resolved comments

ShivamJhaa committed:
Resolved comments

ShivamJhaa committed:
Resolved comments

ShivamJhaa committed:
Resolved comments

ShivamJhaa commented:
@ashutoshc8101  PTAL

DubeySandeep commented on core/templates/components/entity-creation-services/story-creation.service.ts:
```diff
@@ -16,93 +16,73 @@
+    private loaderService: LoaderService,
+    private storyCreationBackendApiService: StoryCreationBackendApiService,
+    private urlInterpolationService: UrlInterpolationService
 
```
extra newline?

ShivamJhaa replied:
Removed

DubeySandeep commented on .github/CODEOWNERS:
```diff
@@ -116,6 +116,7 @@
 /core/domain/collection*.py @aks681
 /core/templates/components/entity-creation-services/collection-creation.service*.ts @EricZLou
 /core/templates/components/entity-creation-services/collection-creation-backend-api.service*.ts @EricZLou
+/core/templates/components/entity-creation-services/story-creation-backend-api.service*.ts @EricZLou
```
This should go inside `story project` section

ShivamJhaa replied:
Done

ShivamJhaa replied:
Done

DubeySandeep reviewed (changes requested):
LGTM (for the only codeowner file CODEOWNERS), left a few minor nit comments PTAL!

ShivamJhaa committed:
nits

ShivamJhaa committed:
lints checks

ShivamJhaa commented:
@DubeySandeep PTAL

srijanreddy98 reviewed (changes requested):
I have no concerns. We need to trim the commit message before merge so I am not approving the PR. Once you have all the approvals and the tests passing, I will merge the PR. Thanks.

EricZLou commented on core/templates/components/forms/custom-forms-directives/edit-thumbnail-modal.component.ts:
```diff
@@ -86,6 +87,7 @@ export class EditThumbnailModalComponent implements OnInit {
       height: Math.round(height),
       width: Math.round(width)
     };
+    this.changeDetectorRef.detectChanges();
```
What does `changeDetectorRef` do and why do we need it here?

ShivamJhaa replied:
It is used to upload the thumbnail image in one click, as the parent is still in ANgularJs, change detection can only be achieved by applying it manually and I have taken help with @ashutoshc8101 and @Radesh-kumar regarding this issue.

EricZLou commented on core/templates/components/entity-creation-services/story-creation.service.ts:
```diff
@@ -16,93 +16,72 @@
+      this.storyCreationBackendApiService.createStoryAsync(
+        newlyCreatedStory, imagesData, bgColor).then((response) => {
+        this.storyCreationInProgress = false;
+        this.imageLocalStorageService.flushStoredImagesData();
```
At this point in the code, do we also `flushStoredImageData` in the original code?

ShivamJhaa replied:
No, updated it now.

EricZLou reviewed (changes requested):
Left some comments for my 2 codeowner files!

DubeySandeep reviewed (approved):
LGTM for the changes in `.github/CODEOWNERS`

aks681 reviewed (commented):
Same comment as @EricZLou. The thumbnail should update on selection and shouldn't require 2 clicks.

ShivamJhaa commented:
> Same comment as @EricZLou. The thumbnail should update on selection and shouldn't require 2 clicks.
This issue is resolved using changeDetectionRef. The video here was not updated, I have updated it now.

ShivamJhaa committed:
Updated as per comments

ShivamJhaa committed:
updated as per comments

ShivamJhaa committed:
updated as per comments

ShivamJhaa committed:
merge conflicts

ShivamJhaa committed:
merge conflicts

ShivamJhaa committed:
Frontend fixed

ShivamJhaa committed:
lints checks

ShivamJhaa committed:
merge conflicts

ShivamJhaa committed:
merge conflicts

aishwary023 commented on core/templates/components/entity-creation-services/story-creation-backend-api.service.spec.ts:
```diff
@@ -0,0 +1,126 @@
+    httpTestingController.verify();
+  });
+
+  it('should successfully create a new story',
```
Generally, we separate the test into three parts:
1. Setup
2. Execution/calling the function
3. Expectations

You may modify the tests to reflect the same, by leaving a new line between the parts.

aishwary023 replied:
Makes the tests much more readable. Please apply the same, here and in other tests throughout the PR. Thanks!

ShivamJhaa replied:
Done.

ShivamJhaa committed:
code cleaning

ShivamJhaa commented:
@aishwary023  PTAL

ShivamJhaa committed:
Merge branch 'develop' of https://github.com/oppia/oppia into migration-story-creation

ShivamJhaa committed:
checks

ShivamJhaa committed:
checks

EricZLou commented on core/templates/components/entity-creation-services/story-creation.service.ts:
```diff
@@ -16,93 +16,72 @@
+import { ImageLocalStorageService } from 'services/image-local-storage.service';
+import { LoaderService } from 'services/loader.service';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
```
Please put any imports that begin with `@` at the top of the list. It may also be a good idea to put a new line after the last `@` import for easier readability!

ShivamJhaa replied:
Done.

EricZLou commented on core/templates/components/entity-creation-services/story-creation.service.ts:
```diff
@@ -16,93 +16,72 @@
+        this.storyCreationInProgress = false;
+        newTab.location.href = this.urlInterpolationService.interpolateUrl(
+          this.STORY_EDITOR_URL_TEMPLATE, {
+            story_id: response.storyId
```
This used to be `response.data.storyId`. Can you make sure it should be changed to `response.storyId`?

ShivamJhaa replied:
I have declared the interface in the story-creation-backend-api file named "StoryCreationResponse" from that we can directly access the storyId, so we don't need to put data in between.

EricZLou reviewed (changes requested):
Some comments -- thanks!

ShivamJhaa committed:
aligned imports

ShivamJhaa commented:
@EricZLou  PTAL
Thanks

EricZLou reviewed (approved):
LGTM -- thanks!

ShivamJhaa committed:
lighthouse tests

ShivamJhaa committed:
lighthouse test

aishwary023 reviewed (approved):
LGTM!

ShivamJhaa committed:
e2e

kevintab95 reviewed (approved):
LGTM for codeowner files. Thanks @ShivamJhaa!

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into migration-story-creation

ShivamJhaa committed:
merege conflicts

ShivamJhaa committed:
e2e

ShivamJhaa committed:
refactor

ShivamJhaa committed:
lighouse checks

ShivamJhaa committed:
lighthouse tests

ShivamJhaa committed:
Merge remote-tracking branch 'upstream/develop' into migration-story-creation

ShivamJhaa committed:
lighthouse tests

ShivamJhaa committed:
e2e
